packages/libs/react-ui/src/components/Accordion/Accordion.tsx

```diff
@@ -1,43 +1,53 @@
 'use client';
 
-import { AccordionSection, IAccordionSectionProps } from './AccordionSection';
+import { IAccordionSectionProps } from '.';
 
-import React, { FC, useState } from 'react';
+import type { FC, FunctionComponentElement } from 'react';
+import React, { useEffect, useState } from 'react';
 
-export interface IAccordionProps {
-  sections: Omit<IAccordionSectionProps, 'isOpen' | 'onToggle'>[];
+export interface IAccordionRootProps {
+  children?: FunctionComponentElement<IAccordionSectionProps>[];
   linked?: boolean;
+  initialOpenSection?: number;
 }
 
-export const Accordion: FC<IAccordionProps> = ({ sections, linked = true }) => {
-  const [expandedSections, setExpandedSections] = useState<number[]>([]);
-  const handleOpen = (index: number): void => {
-    if (linked) setExpandedSections([index]);
-    else setExpandedSections([...expandedSections, index]);
-  };
-  const handleClose = (index: number): void => {
-    setExpandedSections(expandedSections.filter((item) => item !== index));
-  };
+export const AccordionRoot: FC<IAccordionRootProps> = ({
+  children,
+  linked = false,
+  initialOpenSection = undefined,
+}) => {
+  const [openSections, setOpenSections] = useState([initialOpenSection]);
 
-  const isOpen = (index: number): boolean => expandedSections.includes(index);
-
-  const handleToggle = (index: number): void => {
-    if (isOpen(index)) handleClose(index);
-    else handleOpen(index);
-  };
+  useEffect(() => {
+    if (linked && openSections.length > 1) {
+      const lastOpen = openSections.pop() || undefined;
+      setOpenSections([lastOpen]);
+    }
+  }, [linked]);
 
   return (
-    <div data-testid="kda-accordion-wrapper">
-      {sections.map((section, index) => (
-        <AccordionSection
-          {...section}
-          isOpen={isOpen(index)}
-          onToggle={() => handleToggle(index)}
-          key={String(section.title)}
-        >
-          {section.children}
-        </AccordionSection>
-      ))}
+    <div data-testid="kda-accordion-sections">
+      {React.Children.map(children, (section, sectionIndex) =>
+        React.cloneElement(
+          section as React.ReactElement<
+            HTMLElement | IAccordionSectionProps,
+            | string
+            | React.JSXElementConstructor<JSX.Element & IAccordionSectionProps>
```

Comment:
Hmm I feel like you need to check if the `section` is a valid element or I think you'll get an error cloning it. For example what happens if it is a string?

Reply:
Children are typed though, TS should show an error if you pass a string?

Reply:
Well I think currently this is typed to allow a string? Maybe I'm wrong, just looks like it could break when passing in an element that is not an accordion section.

Reply:
The effective type comes from the children as defined in the `IAccordionRootProps` but I understand the confusion as this part isn't necessary anymore. I'll change it to correctly reflect the expected type.